deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/security/EnvStageBodyExtractor.java

```diff
@@ -16,151 +16,26 @@
-                }
-            }
-        }
-
-        if (HotfixBean.class.equals(beanClass) || tryAll) {
-            try {
-                return extractHotfixResource(inputStream);
-            } catch (IOException e) {
-                if (!tryAll) {
-                    throw new BeanClassExtractionException(beanClass, e);
-                }
-            }
-        }
-
-        if (AgentBean.class.equals(beanClass) || tryAll) {
-            try {
-                return extractAgentResource(inputStream);
-            } catch (Exception e) {
-                if (!tryAll) {
-                    if (e instanceof NotFoundException) {
-                        throw (NotFoundException) e;
-                    }
-                    throw new BeanClassExtractionException(beanClass, e);
-                }
-            }
-        }
-
-        if (tryAll) {
-            throw new ExtractionException(
-                    "Failed to extract environment resource using all supported classes");
-        }
-        throw new UnsupportedOperationException("Failed to extract environment resource");
-    }
-
-    private AuthZResource extractEnvironResource(InputStream inputStream) throws IOException {
-        EnvironBean envBean = new ObjectMapper().readValue(inputStream, EnvironBean.class);
-        return new AuthZResource(envBean.getEnv_name(), envBean.getStage_name());
-    }
-
-    private AuthZResource extractDeployResource(InputStream inputStream) throws IOException, ExtractionException, NotFoundException {
-        DeployBean deployBean = new ObjectMapper().readValue(inputStream, DeployBean.class);
-        EnvironBean envBean;
         try {
-            envBean = environDAO.getById(deployBean.getEnv_id());
-        } catch (Exception e) {
-            throw new ExtractionException("Failed to get environment bean", e);
-        }
-        if (envBean == null) {
-            throw new NotFoundException(String.format("Environment(%s) not found, referenced by deploy(%s)",
-                    deployBean.getEnv_id(), deployBean.getDeploy_id()));
-        }
-        return new AuthZResource(envBean.getEnv_name(), envBean.getStage_name());
-    }
-
-    private AuthZResource extractHotfixResource(InputStream inputStream) throws IOException, ExtractionException {
-        HotfixBean hotfixBean = new ObjectMapper().readValue(inputStream, HotfixBean.class);
-        String hotfixId = hotfixBean.getId();
-        try {
-            hotfixBean = hotfixDAO.getByHotfixId(hotfixId);
-        } catch (Exception e) {
-            throw new ExtractionException("Failed to get hotfix bean", e);
-        }
-        if (hotfixBean == null) {
-            throw new NotFoundException(String.format("Hotfix(%s) not found", hotfixId));
-        }
-        return new AuthZResource(hotfixBean.getEnv_name(), "");
-    }
-
-    private AuthZResource extractAgentResource(InputStream inputStream) throws IOException, ExtractionException, NotFoundException {
```

Comment:
i did not find the corresponding extrator class for agent. should we keep this logic around

Reply:
I will reimplement in https://github.com/pinterest/teletraan/pull/1489